autosklearn/ensemble_builder.py

```diff
@@ -671,12 +679,32 @@ def predict(self, set_: str,
         # TODO: ADD saving of predictions on "ensemble data"
 
     def _delete_non_candidate_models(self, candidates):
-        cand_ids = [cand.replace('_', '.').split('.')[-2] for cand in candidates]
-        for model_file in self.read_preds.keys():
-            model_file_id = model_file.split('.')[-2]
-            if model_file_id not in cand_ids:
-                os.remove(model_file)
-                self.logger.info("Deleted file of non-candidate model %s", model_file)
+        for pred_file in self.read_preds.keys():
+            if pred_file not in candidates and pred_file not in self.deleted_preds:
+                # Delete prediction file
```

Comment:
Could you please add an additional check whether there are files stored for a potential external validation and test set and delete them as well? They should be in the directories `predictions_valid` and `preditions_test`.

Reply:
Sure. But could you please tell me under which circumstances are those files created? Is there a test/example to which I could refer to?
When I run the test_delete_non_candidate_models, no such files are created.

Reply:
They are created if you pass an additional `X_test` and `y_test` to the `AutoSklearnClassifier.fit()` method. You can see how they're loaded in the method `EnsembleBuilder.get_valid_test_preds()`.

Reply:
I tried adding `X_test` and `y_test` and the `prediction_test` folder was created, but not the `prediction_valid`.  Am I missing something?

Reply:
>  Am I missing something?

No, actually not. Getting `prediction_valid` to be written to disk is not possible with the standard interface, so there's no need for tests for that.